mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java

```diff
@@ -77,6 +80,9 @@ private ObjectMapper initializeObjectMapper() {
 
     @Override
     public ParsedContent extractContent(InputStream inputStream, String contentType) throws Exception {
+        if (contentType.startsWith("text/")) {
+            return jsoupTextExtractor.extractContent(inputStream, contentType);
```

Comment:
not only for html?

Reply:
Why wiuld we pay the tika price for text?

Reply:
is jsoup able to handle text only?

Reply:
you can simply ignore the extract if it is only text, and avoid paying the cost for jsoup also :)

Reply:
Well, `you can simply ignore the extract if it is only text,` will result in not getting any text for `text/plain`... Not what we want.

`avoid paying the cost for jsoup also` JsoupTextExtractor parses html only for `text/html` and do avoid html parsing on non HTML document, thus is simply returning the underlying text for non html text.